File: php/features.php
<?php
declare(strict_types=1);

// IpAddressLookup SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpAddressLookupFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpAddressLookupBaseFeature();
            case "test":
                return new IpAddressLookupTestFeature();
            default:
                return new IpAddressLookupBaseFeature();
        }
    }
}

File: py/features.py
# IpAddressLookup SDK feature factory

from feature.base_feature import IpAddressLookupBaseFeature
from feature.test_feature import IpAddressLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpAddressLookupBaseFeature(),
        "test": lambda: IpAddressLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
